stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/imagepreview/ImagePreviewActivity.kt

```diff
@@ -73,26 +133,79 @@ public class ImagePreviewActivity : AppCompatActivity() {
 
-                ImagePreviewContentWrapper(message)
+                if (message.id.isNotEmpty()) {
+                    ImagePreviewContentWrapper(message, attachmentPosition)
+                }
             }
         }
     }
 
+    /**
+     * Wraps the content of the screen in a composable that represents the top and bottom bars and the
+     * images preview.
+     *
+     * @param message The message to show the attachments from.
+     * @param initialAttachmentPosition The initial pager position, based on the image the user clicked on.
+     */
+    @OptIn(ExperimentalAnimationApi::class)
     @Composable
-    private fun ImagePreviewContentWrapper(message: Message) {
+    private fun ImagePreviewContentWrapper(
+        message: Message,
+        initialAttachmentPosition: Int,
+    ) {
+        val pageCount = message.attachments.size
+        val startingPosition = if (initialAttachmentPosition >= pageCount) 0 else initialAttachmentPosition
+
         val pagerState = rememberPagerState(
-            pageCount = message.attachments.size,
+            pageCount = pageCount,
+            initialPage = startingPosition,
             initialOffscreenLimit = 2
         )
 
-        Scaffold(
-            modifier = Modifier.fillMaxSize(),
-            topBar = { ImagePreviewTopBar(message) },
-            content = { ImagePreviewContent(pagerState, message) },
-            bottomBar = { ImagePreviewBottomBar(message, pagerState) }
-        )
+        Box(modifier = Modifier.fillMaxSize()) {
+            Scaffold(
+                modifier = Modifier.fillMaxSize(),
+                topBar = { ImagePreviewTopBar(message) },
+                content = { ImagePreviewContent(pagerState, message.attachments) },
+                bottomBar = { ImagePreviewBottomBar(message.attachments, pagerState) }
+            )
+
+            AnimatedVisibility(
+                visible = imagePreviewViewModel.isShowingOptions,
+                enter = fadeIn(),
+                exit = fadeOut()
+            ) {
+                ImagePreviewOptions(
+                    options = defaultImageOptions(message = message),
+                    pagerState = pagerState,
+                    modifier = Modifier.animateEnterExit(
+                        enter = slideInVertically(),
+                        exit = slideOutVertically()
+                    )
+                )
+            }
+
+            AnimatedVisibility(
+                visible = imagePreviewViewModel.isShowingGallery,
+                enter = fadeIn(),
+                exit = fadeOut()
+            ) {
+                ImageGallery(
+                    pagerState = pagerState,
+                    modifier = Modifier.animateEnterExit(
+                        enter = slideInVertically(initialOffsetY = { height -> height / 2 }),
+                        exit = slideOutVertically(targetOffsetY = { height -> height / 2 })
+                    )
+                )
+            }
```

Comment:
Great animations here!

Reply:
Thanks! It's just basic stuff that's default but flipped over for our use case!

However, I'm starting to add more of these to make the basic interactions a bit nicer ^^